tests/marine/test_marine.py

```diff
@@ -1025,3 +1025,34 @@ def test_parse_fields_preserves_order(marine_instance: Marine, tcp_packet: bytes
         "udp.srcport": None,
         "tcp.srcport": "16424",
     }
+
+
+def test_parse_all_fields_int_value(tcp_packet_fields):
+    tcp_source_port = tcp_packet_fields["tcp"]["tcp.srcport"]
+    assert isinstance(tcp_source_port, int)
+    assert tcp_source_port == 16424
+
+
+def test_parse_all_fields_str_value(tcp_packet_fields):
+    ip_src = tcp_packet_fields["ip"]["ip.src"]
+    assert isinstance(ip_src, str)
+    assert ip_src == "10.0.0.255"
+
+
+def test_parse_all_fields_list_value(tcp_packet_fields):
+    ip_addr = tcp_packet_fields["ip"]["ip.addr"]
+    assert isinstance(ip_addr, list)
+    assert "10.0.0.255" in ip_addr
```

Comment:
If you know exactly what the list looks like, consider comparing it directly to the list
```python
assert ["10.0.0.255", "21.53.78.255"] == ip_addr
```

Reply:
I cannot know ahead of time the order of the list produced by the dissector, So I though checking it that way would be safer.